src/providers/twitch/TwitchIrcServer.cpp

```diff
@@ -296,6 +302,64 @@ std::shared_ptr<Channel> TwitchIrcServer::getChannelOrEmptyByID(
     return Channel::getEmpty();
 }
 
+namespace {
+    // TODO: combine this with getEmoteSetBatches in TwitchAccount.cpp, maybe some templated thing
+    std::vector<QStringList> getChannelsInBatches(QStringList channels)
+    {
+        constexpr int batchSize = 100;
+
+        int batchCount = (channels.size() / batchSize) + 1;
+
+        std::vector<QStringList> batches;
+        batches.reserve(batchCount);
+
+        for (int i = 0; i < batchCount; i++)
+        {
+            QStringList batch;
+
+            // I hate you, msvc
+            int last = (std::min)(batchSize, channels.size() - batchSize * i);
+            for (int j = 0; j < last; j++)
+            {
+                batch.push_back(channels.at(j + (batchSize * i)));
+            }
+            batches.emplace_back(batch);
+        }
+
+        return batches;
+    }
+}  // namespace
+
+void TwitchIrcServer::bulkRefreshLiveStatus()
+{
+    QStringList userIDs;
+    this->forEachChannel([&userIDs](ChannelPtr chan) {
+        auto twitchChan = dynamic_cast<TwitchChannel *>(chan.get());
+        if (!twitchChan->roomId().isEmpty())
+            userIDs.push_back(twitchChan->roomId());
+    });
+
+    for (const auto &batch : getChannelsInBatches(userIDs))
+    {
+        getHelix()->fetchStreams(
+            batch, QStringList(),
+            [this](std::vector<HelixStream> streams) {
+                for (const auto &stream : streams)
+                {
+                    auto chan = this->getChannelOrEmptyByID(stream.userId);
```

Comment:
Consider using `getChannelOrEmpty` in twitch irc server on `stream.userLogin`

* `getChannelOrEmptyByID` is an O(n) operation while `getChannelOrEmpty` is O(1)
* channel id's aren't populated until ROOMSTATE is received, which may be some time if trying to join many channels on start up (so the current approach may not mark some channels as live that we otherwise could)

Reply:
Now you can submit a PR ;)

Reply:
> channel id's aren't populated until ROOMSTATE is received, which may be some time if trying to join many channels on start up (so the current approach may not mark some channels as live that we otherwise could)

This logic isn't used when we initially load channels - at first launch we still perform the burst of helix requests for every channel separetly.